hooks/presence.py

```diff
@@ -0,0 +1,152 @@
+        ConnectionRefusedError,
+        InvalidID,
+        InvalidPipe,
+        FileNotFoundError,
+        ConnectionResetError
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._pid = os.getpid()
+
+        self._details = {
+            'details': "Regretting...",
+            'state': None,
+            'start': None,
+            'img': None,
+            'txt': None
+        }
+
+    def present(self, engine, start=None, details="Regretting...", state=None):
+        """
+        Set status for DiscordRPC.
+        """
+        self._details = {
+            'details': details,
+            'state': state,
+            'start': time.time()*1000 - start if start else None,
+            'img': engine.account["api"],
+            'txt': "{} at {}".format(
+                engine.account["username"],
+                engine.account["api"]
+            )
+        }
+        self._update = True
+
+    def run(self):
+        set_loop(new_loop())
+        while True:
+            try:
+                self._reconnect()
+                if self._enabled:
+                    if self._update:
+                        self._rpc.set_activity(
+                            pid=self._pid,
+                            large_image="icon",
+                            large_text=self._details['details'],
+                            small_image=self._details['img'],
+                            small_text=self._details['txt'],
+                            details=self._details['details'],
+                            state=self._details['state'],
+                            start=self._details['start']
+                        )
+                        self._update = False
+                    time.sleep(1)
+            except self._errors:
+                self._enabled = False
+                try:
+                    self._rpc.close()
+                except AttributeError:
+                    pass
+
+    def _reconnect(self):
+        if not self._enabled:
+            try:
+                self._rpc = Client(self._client_id)
+                self._rpc.start()
+                self._enabled = True
+
+            except self._errors:
+                self._enabled = False
+
+
+rpc = DiscordRPC(daemon=True)
+
+
+def init(engine):
+    """
+    Initialize this hook.
+    """
+    rpc.start()
+    print('a')
+    rpc.present(engine)
+
+
+def tracker_state(engine, status):
+    """
+    Update status in thread.
+    """
+    if status["state"] == 1:
+        show = status["show"][0]
+        title = show["titles"][0]
+        episode = status["show"][-1]
+        total = show["total"] or estimate_aired_episodes(
+            engine.get_show_info(show['id'])
+        ) or '?'
+
+        if status["paused"]:
+            rpc.present(engine,
+                        status["viewOffset"],
+                        "Paused {}".format(title),
+                        "Episode {} of {}".format(
```

Comment:
I've noticed this while rebasing my other PR earlier, but does this repo use/enforce 80 character max width and do we think this is contemporary? I personally prefer going up to 120 instead of wrapping eagerly like this.

Reply:
I prefer to follow pep-8 style guideline.